client/network/common/src/sync.rs

```diff
@@ -395,4 +395,14 @@ pub trait ChainSync<Block: BlockT>: Send {
 
 	/// Decode implementation-specific state response.
 	fn decode_state_response(&self, response: &[u8]) -> Result<OpaqueStateResponse, String>;
+
+	/// Advance the state of `ChainSync`
+	///
+	/// Internally calls [`ChainSync::poll_block_announce_validation()`] and
+	/// this function should be polled until it returns [`Poll::Pending`] to
+	/// consume all pending events.
```

Comment:
Hmm, this comment is "weird". When `poll` is already calling `poll_block_announce_validation`, this function shouldn't exist in the trait from the beginning. 

I would also maybe call this `poll_next`. For `poll` the contract is more, "poll me once" and check the result. While this here is more `Stream` like.

Reply:
I will remove the function and rename the other one to `poll_next()`